apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,301 @@
+'use strict';
+/* global ApplicationSource */
+/* global BookmarkSource */
+/* global CollectionSource */
+/* global configurator */
+/* global dispatchEvent */
+/* global GaiaGrid */
+
+(function(exports) {
+
+
+	/**
+	 * This function as a constructor of PinAppNavigation. It gets dom element that contains elements
+	 * @param {Object} list [Container that contains pin elements]
+	 */
+	function PinAppNavigation(list){
+
+		if (!list &&  typeof list !== 'object'){
+			console.log('List of pin apps must be an boject');
+			return;
+		}
+
+		this.elemList = list;
+		window.removeEventListener('keydown', this);
+		window.addEventListener('keydown', this);
+
+	};
+
+
+	PinAppNavigation.prototype = {
+
+		selector: '',
+
+		elements: null,
+
+		preventDef: false,
+
+		selectedElemIndex: 0,
+
+		elemList: null,
+
+		prevVerticalKey: null,
+
+
+		/**
+		 * This function reset all pinns to default vlue.
+		 * @return {[nothing]} [nothing]
+		 */
+		reset: function reset(){
+
+			this.elements.sort(function(elem1, elem2){
+				if (!elem1.dataset.index || !elem2.dataset.index){
+					return 1;
+				}
+				return elem1.dataset.index - elem2.dataset.index;
+			});
+
+
+			var selected = this.elemList.querySelector('.selected');
+
+			if (selected){
+				selected.classList.remove('selected')
+			}
+
+			this.selectedElemIndex = 0;
+
+			for (var elemIndex in this.elements){
+				this.elemList.appendChild(this.elements[elemIndex]);
+			}
+
+			this.elements[this.selectedElemIndex].classList.add('selected');
+			this.elemList.removeAttribute('style');
+
+			var toRemoved = this.elemList.querySelector('.removed');
+			var toRemoveOutFocus = this.elemList.querySelector('.out-focus');
+
+			if (toRemoved){
+				this.elemList.removeChild(toRemoved);
+			}
+			toRemoveOutFocus ? toRemoveOutFocus.classList.remove('out-focus') : null;
+
+			window.removeEventListener('keydown', this);
+			window.addEventListener('keydown', this);
+
+		},
+
+
+
+		/**
+		 * This function refresh elements by selector
+		 * @return {[nothins]} [nothing]
+		 */
+		refresh: function refresh(){
+
+				if (!this.selector){
+					console.log('Can not work without selector');
+					return;
+				}
+
+				this.elements = Array.prototype.slice.call(document.querySelectorAll(this.selector));
+
+				if (!this.elements && !this.elements.length){
+					cnosole.log('Can not find any elements by this selector');
```

Comment:
I know what console.log() does. "typo" means misprint or literal error. In this particular line you will not see your logs, because you typed cNosole instead of console :)